src/java.base/windows/native/libnio/ch/UnixDomainSockets.c

```diff
@@ -87,11 +87,27 @@ jint unixSocketAddressToSockaddr(JNIEnv *env, jbyteArray addr, struct sockaddr_u
 JNIEXPORT jboolean JNICALL
 Java_sun_nio_ch_UnixDomainSockets_socketSupported(JNIEnv *env, jclass cl)
 {
+    OSVERSIONINFOEX osver;
     SOCKET s = socket(PF_UNIX, SOCK_STREAM, 0);
     if (s == INVALID_SOCKET) {
         return JNI_FALSE;
     }
     closesocket(s);
+
+    osver.dwOSVersionInfoSize = sizeof(osver);
+
+    /* GetVersionEx is deprecated, but no suitable other API exists yet */
+#pragma warning(suppress : 4996)
+    if (!GetVersionEx((OSVERSIONINFO *)&osver)) {
+        return JNI_FALSE;
+    }
+    /*
+     * 18362 is the lowest OS build number known to not have bug
+     * where socket files can not be opened using normal win32 open flags
+     */
+    if (osver.dwBuildNumber < 18362) {
+        return JNI_FALSE;
+    }
```

Comment:
Can you use VerifyVersionInfoW instead of the deprecated GetVersionEx? The IsWindows10RS3OrGreater function in net_util_md.c is a recent example.

Reply:
It looks like VerifyVersionInfoW is (or will be) deprecated also and it seems like quite a complicated API that checks the same thing in this case.

https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-verifyversioninfow

Reply:
I'm going to close this PR as it should have been on the jdk16 repo. I'll also look at using the other Windows API that does not require a #pragma (for now).